x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/indexlifecycle/ExecuteStepsUpdateTaskTests.java

```diff
@@ -170,19 +174,21 @@ public void testExecuteUntilFirstNonClusterStateStep() throws IOException {
         assertThat(LifecycleSettings.LIFECYCLE_STEP_INFO_SETTING.get(newState.metaData().index(index).getSettings()), equalTo(""));
     }
 
+    // TODO: I think these tests are invalid now that policy name is not used for retrieving the step
```

Comment:
I'd love to hear feedback about these tests, whether they should be removed or what?

Reply:
hmm. I think part of the problem is that I do not think we need a `firstStepMap` anymore? Since we can see our index is just initializing when we update the policy-registry and seed the `indexPhaseSteps` with the Steps from the first phase and just retrieve the first item in the list from there when asked for `getFirstStep`.

Reply:
I'll investigate removing the `firstStepMap`, good idea :)

Reply:
After investigating this, removing `firstStepMap` is much harder than it looks, so I'll leave that for a subsequent PR